types/src/checksummed_hex.rs

```diff
@@ -82,18 +83,18 @@ pub fn encode_iter(input: &(impl AsRef<[u8]> + ?Sized)) -> impl Iterator<Item =
 /// Returns true if all chars in a string are uppercase or lowercase.
 /// Returns false if the string is mixed case or if there are no alphabetic chars.
 fn string_is_same_case<T: AsRef<[u8]> + ?Sized>(s: &T) -> bool {
-    let lower_range = 97..=102u8; // a..=f
-    let upper_range = 65..=70u8; // A..=F
+    const LOWER_RANGE: RangeInclusive<u8> = 'a' as u8..='f' as u8;
+    const UPPER_RANGE: RangeInclusive<u8> = 'A' as u8..='F' as u8;
```

Comment:
I'm not an expert in Rust optimization but this looks like it could be extract to be a `const` outside of this function _and_ have the "containst" methods also be `const`. What do you think?

Reply:
Nice! Thanks for taking the time to explain!